aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/RequestLogger.java

```diff
@@ -1,109 +1,123 @@
+            logMessage.put(MESSAGE, throwable.getClass().getCanonicalName());
+            logMessage.put(throwable);
+            logMessage.put(requestDataMap);
+            logMessage(logMessage.toString());
+        } catch (Throwable caughtThrowable) {
+            logMessage(caughtThrowable);
+        }
+    }
+
     public void logAndThrow(Throwable throwable) {
         log(throwable);
         ExceptionUtils.rethrow(throwable);
     }
 
-    public <RequestT, ResponseT, ClientT> BiFunction<RequestT, ProxyClient<ClientT>, ResponseT> log(
-            final BiFunction<RequestT, ProxyClient<ClientT>, ResponseT> makeServiceCallMethod) {
-        return (request, proxyClient) -> {
-            ResponseT result = null;
-            try {
-                log("RdsRequest", request, ImmutableMap.of("Operation", request.getClass().getSimpleName()));
-                result = makeServiceCallMethod.apply(request, proxyClient);
-                log("RdsResponse", result, ImmutableMap.of("Operation", result.getClass().getSimpleName()));
-            } catch (Exception e) {
-                logAndThrow(e);
-            }
-            return result;
-        };
-    }
-
-    public void log(Throwable throwable) {
+    public void log(Object object) {
         try {
-            JSONObject jsonObject = new JSONObject(jsonPrinter.print(throwable));
-            jsonObject.put(REQUEST_DATA_MARKER, requestData);
-            logMessage(jsonObject.toString());
-        } catch (Throwable caughtThrowable) {
+            DEFAULT_LOG_RULE_SET.handle(object);
+        } catch (Throwable throwable) {
             logMessage(throwable);
         }
     }
 
-    public void log(String message) {
-        log(MESSAGE_MARKER, message);
-    }
-
-    public void log(String marker, Object object) {
-        log(marker, object, null);
+    public void log(String message, Object object) {
+        log(message, object, null);
     }
 
-    public void log(String marker, Object object, Map<String, String> additionalFields) {
+    public void log(String message, Object object, Map<String, String> additionalFields) {
         try {
-            String objectAsString = jsonPrinter.print(ObjectUtils.defaultIfNull(object, StringUtils.EMPTY));
-            final JSONObject jsonLog = new JSONObject();
-            jsonLog.put(marker, new JSONObject(objectAsString));
-            Optional.ofNullable(additionalFields).orElse(Collections.emptyMap())
-                    .entrySet().stream().forEach(entry -> jsonLog.put(entry.getKey(), entry.getValue()));
-            jsonLog.put(REQUEST_DATA_MARKER, requestData);
-            logMessage(jsonLog.toString());
+            LogMessage logMessage = JsonLogMessage.newLogMessage(jsonPrinter);
+            logMessage.put(MESSAGE, message);
+            logMessage.put(object);
+            logMessage.put(additionalFields);
+            logMessage.put(requestDataMap);
+            logMessage(logMessage.toString());
```

Comment:
This line could be improved. Firstly, the name collision: it looks confusing. Secondly, logMessage.toString() is worth an overloaded method `logMessage(final LogMessage message) { // call .toString() explicitly here }`: otherwise we're exposing too much of an implementation details.

Reply:
Done. Thanks for suggesting!